splice_machine/src/main/java/com/splicemachine/derby/utils/Vacuum.java

```diff
@@ -199,6 +199,9 @@ public void vacuumDatabase(long oldestActiveTransaction) throws SQLException{
 
             List<DataDelete> deletes = new ArrayList<>();
             for (long removed : toDelete) {
+                boolean restoreMode = SIDriver.driver().lifecycleManager().isRestoreMode();
+                if (restoreMode)
+                    return;
```

Comment:
On a second thought, I think this code may have an issue. When we get here, all the eligible dropped tables have been deleted, and this FOR loop is just to prepare for the deletion of the corresponding entries in the DROPPED_CONGLOMERATES_TABLE. If we abort here, the tables deleted will never get a chance to be removed from the DROPPED_CONGLOMERATES_TABLE. Correct me if I missed something.

Reply:
That's right @yxia92 , we want to break out of the "for(TableDescriptor table:hTableDescriptors){" loop, not this one.